integrations/go/workload_executor_test.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+	"go.mongodb.org/mongo-driver/mongo/integration/unified"
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+)
+
+// hasLoop returns whether or not testCase tc contains a loop operation
+func hasLoop(tc *unified.TestCase) bool {
+	if tc == nil {
+		return false
+	}
+	for _, op := range tc.Operations {
+		if op.Name == "loop" {
+			return true
+		}
+	}
+	return false
+}
+
+// marshalStructToFile marshals the given object and writes to filePath
+func marshalStructToFile(t *testing.T, obj interface{}, filePath string) {
+	t.Helper()
+	// MarshalExtJSON is used to print out bson.Raw properly
+	marshaled, err := bson.MarshalExtJSON(obj, false, false)
```

Comment:
Minor, but should this instead be canonical? I don't see a reference in the workload executor specification as to whether this should be canonical or relaxed.

```suggestion
	marshaled, err := bson.MarshalExtJSON(obj, true, false)
```

Reply:
The output should be json, not extended json.

Reply:
Yeah, the relaxed extended json ends up being identical to json for results.json, and for events.json actually produces human readable events documents.